src/hmac.rs

```diff
@@ -0,0 +1,130 @@
+use crate::callback::CallbackSubscription;
+use crate::callback::Consumer;
+use crate::result::TockResult;
+use crate::syscalls;
+use core::marker::PhantomData;
+use libtock_core::shared_memory::SharedMemory;
+
+const DRIVER_NUMBER: usize = 0x40003;
+
+pub const KEY_BUFFER_SIZE: usize = 32;
+pub const DATA_BUFFER_SIZE: usize = 256;
+pub const DEST_BUFFER_SIZE: usize = 32;
+
+mod command_nr {
+    pub const SET_ALGORITHM: usize = 0;
+    pub const RUN: usize = 1;
+}
+
+mod subscribe_nr {
+    pub const SUBSCRIBE_CALLBACK: usize = 0;
+}
+
+mod allow_nr {
+    pub const KEY: usize = 0;
+    pub const DATA: usize = 1;
+    pub const DEST: usize = 2;
+}
+
+#[non_exhaustive]
+pub struct HmacDriverFactory;
+
+impl HmacDriverFactory {
+    pub fn init_driver(&mut self) -> TockResult<HmacDriver> {
+        let hmac = HmacDriver {
+            lifetime: PhantomData,
+        };
+        Ok(hmac)
+    }
+}
+
+struct HmacEventConsumer;
+
+impl<CB: FnMut(usize, usize)> Consumer<CB> for HmacEventConsumer {
+    fn consume(callback: &mut CB, result: usize, digest: usize, _: usize) {
+        callback(result, digest);
+    }
+}
+
+pub struct HmacKeyBuffer {
+    // TODO: make this generic if possible with the driver impl
```

Comment:
I would either
 * remove this todo
 * create a github issue for this.
Todos in the code have a high probability to stay there forever.

Reply:
I think the question of "do we allow TODOs in the libtock-rs codebase" is worthy of a discussion outside this PR.

Reply:
I copied the TODO from the ADC code